src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -758,6 +761,8 @@ GenerateCommonEquivalence(List *attributeEquivalenceList,
 		 */
 		if (bms_is_member(currentEquivalenceClass->equivalenceId, addedEquivalenceIds))
 		{
+			++equivalenceClassIndex;
```

Comment:
I always think of post-increment more readable than pre-increment.  We mostly preferred post-increment in our codebase.